crates/miden-objects/src/block/proven_block.rs

```diff
@@ -105,18 +84,23 @@ impl ProvenBlock {
     /// Each note is accompanied by a corresponding index specifying where the note is located
     /// in the block's [`BlockNoteTree`].
     pub fn output_notes(&self) -> impl Iterator<Item = (BlockNoteIndex, &OutputNote)> {
-        self.output_note_batches.iter().enumerate().flat_map(|(batch_idx, notes)| {
-            notes.iter().map(move |(note_idx_in_batch, note)| {
-                (
-                    // SAFETY: The proven block contains at most the max allowed number of batches
-                    // and each batch is guaranteed to contain at most the
-                    // max allowed number of output notes.
-                    BlockNoteIndex::new(batch_idx, *note_idx_in_batch)
-                        .expect("max batches in block and max notes in batches should be enforced"),
-                    note,
-                )
+        self.body
+            .output_note_batches()
+            .iter()
+            .enumerate()
+            .flat_map(|(batch_idx, notes)| {
+                notes.iter().map(move |(note_idx_in_batch, note)| {
+                    (
+                        // SAFETY: The proven block contains at most the max allowed number of
+                        // batches and each batch is guaranteed to contain
+                        // at most the max allowed number of output notes.
+                        BlockNoteIndex::new(batch_idx, *note_idx_in_batch).expect(
+                            "max batches in block and max notes in batches should be enforced",
+                        ),
+                        note,
+                    )
+                })
             })
```

Comment:
It feels like this should be a method on `BlockBody`. This way, here we'd be able to do just:
```rust
self.body.output_notes()
```